warning STL4036: <ciso646> is removed in C++20

We are getting this with a MSVC project using C++20/permissive-:

2>C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.35.32215\include\ciso646(15,47): error C2220: the following warning is treated as an error
2>C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.35.32215\include\ciso646(15,47): warning C4996: '_Header_ciso646': warning STL4036: <ciso646> is removed in C++20. You can define _SILENCE_CXX20_CISO646_REMOVED_WARNING or _SILENCE_ALL_CXX20_DEPRECATION_WARNINGS to suppress this warning.

We do suppress it for now, but it would be nice if the libpq(xx) headers could check for current settings and not include this, if it's not needed.

I found something similar in another repo, and they fixed it somehow (didn't check the details):
https://github.com/nlohmann/json/issues/2089

--- a/NEWS
+++ b/NEWS
@@ -25,6 +25,7 @@
  - Don't use `std::function` as deleter for smart pointers.
  - Work around gcc compile error with regex + address sanitizer + analyzers.
  - Fix "double free" on exit when built as shared library on Debian. (#681)
+ - Stop including `<ciso646>`; should be built into compilers. (#680)
 7.7.4
  - `transaction_base::for_each()` is now called `for_stream()`. (#580)
  - New `transaction_base::for_query()` is similar, but non-streaming. (#580)

--- a/include/pqxx/internal/header-pre.hxx
+++ b/include/pqxx/internal/header-pre.hxx
@@ -52,13 +52,6 @@
 // Workarounds & definitions that need to be included even in library's headers
 #include "pqxx/config-public-compiler.h"
 
-// C++20: No longer needed.
-// Enable ISO-646 alternative operaotr representations: "and" instead of "&&"
-// etc. on older compilers.  C++20 removes this header.
-#if __has_include(<ciso646>)
-#  include <ciso646>
-#endif
-
 
 #if defined(PQXX_HAVE_GCC_PURE)
 /// Declare function "pure": no side effects, only reads globals and its args.

